types/src/transaction/metadata.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This file implements Libra transaction metadata types to allow
+//! easy parsing and introspection into metadata, whether the transaction
+//! is using regular subaddressing, is subject to travel rule or corresponds
+//! to an on-chain payment refund.
+
+use serde::{Deserialize, Serialize};
+
+/// List of all supported metadata types
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum MetadataType {
+    Undefined,
+    GeneralMetadataType(GeneralMetadata),
+    TravelRuleMetadataType(TravelRuleMetadata),
+    UnstructuredStringMetadataType(UnstructuredStringMetadata),
```

Comment:
Can we just have StringMetadataType(String) and BytesMetadataType(Vec[u8])  cause this is actually two different stuff.

Reply:
I'll just put UnstructuredBytesMetadataType, since this doesn't conform to subaddressing or travel rule, it's very likely that the sender (and hopefully the receiver) know how to interpret those bytes in a custom manner - so no need to come up with a string format until we have a need for it (Memos should probably be explicited at the structure level for example).